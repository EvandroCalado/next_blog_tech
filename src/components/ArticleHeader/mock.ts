export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
