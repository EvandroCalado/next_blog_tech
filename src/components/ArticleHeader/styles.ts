import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
