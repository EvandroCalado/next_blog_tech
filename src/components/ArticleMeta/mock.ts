export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
