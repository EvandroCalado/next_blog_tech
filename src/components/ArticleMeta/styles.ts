import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
