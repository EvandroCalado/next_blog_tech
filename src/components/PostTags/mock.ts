export default {
  tags: [
    {
      displayName: 'api',
      slug: 'api',
    },
    {
      displayName: 'Node',
      slug: 'node',
    },
    {
      displayName: 'Javascript',
      slug: 'javascript',
    },
  ],
};
