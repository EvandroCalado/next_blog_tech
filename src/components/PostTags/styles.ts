import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.large} 0;

    span:not(:last-child)::after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
