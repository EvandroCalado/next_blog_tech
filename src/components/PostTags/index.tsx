import Link from 'next/link';
import { Tags } from '../shared-types/tags';
import * as Styled from './styles';

export type PostTagsProps = {
  tags?: Tags[];
};
const PostTags = ({ tags = [] }: PostTagsProps) => {
  if (tags.length === 0) {
    return null;
  }

  return (
    <Styled.Wrapper>
      Tags:{' '}
      {tags.map((tag) => (
        <span key={tag.id}>
          <Link href={`/tag/${tag.slug}`}>{tag.displayName}</Link>
        </span>
      ))}
    </Styled.Wrapper>
  );
};

export default PostTags;
