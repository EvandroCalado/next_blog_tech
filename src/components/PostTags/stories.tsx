import PostTags, { PostTagsProps } from '.';
import { Meta, Story } from '@storybook/react/types-6-0';

import mock from './mock';

export default {
  title: 'PostTags',
  component: PostTags,
  args: mock,
} as Meta;

export const Template: Story<PostTagsProps> = (args) => (
  <div>
    <PostTags {...args} />
  </div>
);

export const NoTags: Story<PostTagsProps> = (args) => (
  <div>
    <PostTags {...args} tags={undefined} />
  </div>
);
