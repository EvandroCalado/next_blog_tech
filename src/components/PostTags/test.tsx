import PostTags from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<PostTags />', () => {
  it('should render tags ', () => {
    renderTheme(<PostTags {...mock} />);
    expect(screen.getByText(/Tags:/i)).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(3);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostTags {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render match snapshot with no tags', () => {
    const { container } = renderTheme(<PostTags {...mock} tags={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
