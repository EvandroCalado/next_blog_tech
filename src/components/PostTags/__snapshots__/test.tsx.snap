// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should match snapshot 1`] = `
.c0 {
  margin: 3.2rem 0;
}

.c0 span:not(:last-child)::after {
  content: ', ';
}

.c0 a {
  color: #f24b4b;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <p
    class="c0"
  >
    Tags:
     
    <span>
      <a
        href="/tag/api"
      >
        api
      </a>
    </span>
    <span>
      <a
        href="/tag/node"
      >
        Node
      </a>
    </span>
    <span>
      <a
        href="/tag/javascript"
      >
        Javascript
      </a>
    </span>
  </p>
</div>
`;

exports[`<PostTags /> should render match snapshot with no tags 1`] = `<div />`;
